File: Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int carryout = 1;
      
        list<int> resultlist;
        for(int i = digits.size()-1; i>-1 ; i--)
        {
            int val = (digits[i]+carryout)%10;
            resultlist.push_front(val);
            carryout = (digits[i]+carryout)/10;
        }
        if(carryout!=0)
            resultlist.push_front(carryout);
            
        vector<int> result(resultlist.begin(), resultlist.end());
        return result;
    }
};

File: Plus One.java
public class Solution {
    public int[] plusOne(int[] digits) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int carryOut = 1;
        int []ret = new int[digits.length+1];
        for(int i = digits.length-1; i > -1; i--)
        {
            int val = digits[i] + carryOut;
            digits[i] = val%10;
            ret[i+1] = digits[i];
            carryOut = val/10;
        }
        if(carryOut==1)
        {
            ret[0] = carryOut;
            return ret;
        }
        else
        {
            return digits;
        }
    }
}